src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -0,0 +1,276 @@
+import json
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+from snowflake.cli.api.console import cli_console as cc
+from snowflake.cli.api.project.schemas.native_app.native_app import NativeApp
+from snowflake.cli.api.project.schemas.native_app.path_mapping import (
+    PathMapping,
+    Processor,
+)
+from snowflake.cli.api.utils.rendering import jinja_render_from_file
+from snowflake.cli.plugins.nativeapp.artifacts import (
+    is_glob,
+    resolve_without_follow,
+)
+from snowflake.cli.plugins.nativeapp.codegen.artifact_processor import (
+    ArtifactProcessor,
+    MissingProjectDefinitionPropertyError,
+)
+from snowflake.cli.plugins.nativeapp.codegen.sandbox import (
+    ExecutionEnvironmentType,
+    SandboxExecutionError,
+    execute_script_in_sandbox,
+)
+from snowflake.cli.plugins.nativeapp.codegen.snowpark.extension_function_utils import (
+    _enrich_entity,
+)
+from snowflake.cli.plugins.nativeapp.utils import (
+    filter_files,
+    get_all_file_paths_under_dir,
+    is_python_file,
+)
+
+DEFAULT_TIMEOUT = 30
+
+
+def _determine_virtual_env(processor: Processor) -> Dict[str, Any]:
+    """
+    Determines a virtual environment to run the Snowpark processor in, either through the project definition or by querying the current environment.
+    """
+    if (processor.properties is None) or ("env" not in processor.properties):
+        return {}
+
+    env_props = processor.properties["env"]
+    env_type = env_props.get("type", None)
+
+    if env_type is None:
+        return {}
+
+    if env_type.upper() == ExecutionEnvironmentType.CONDA.name:
+        env_name = env_props.get("name", None)
+        if env_name is None:
+            raise MissingProjectDefinitionPropertyError(
```

Comment:
This seems odd. Why raise an exception that reads like a warning? Not specifying the name should be fine, if we can't find an env we'll just bail out anyway.

Reply:
I can turn this into a warning instead.